src/validators/arguments_v3.rs

```diff
@@ -0,0 +1,654 @@
+        )
+    }
+}
+
+#[derive(Debug)]
+pub struct ArgumentsV3Validator {
+    parameters: Vec<Parameter>,
+    positional_params_count: usize,
+    loc_by_alias: bool,
+    extra: ExtraBehavior,
+    validate_by_alias: Option<bool>,
+    validate_by_name: Option<bool>,
+}
+
+impl BuildValidator for ArgumentsV3Validator {
+    const EXPECTED_TYPE: &'static str = "arguments-v3";
+
+    fn build(
+        schema: &Bound<'_, PyDict>,
+        config: Option<&Bound<'_, PyDict>>,
+        definitions: &mut DefinitionsBuilder<CombinedValidator>,
+    ) -> PyResult<CombinedValidator> {
+        let py = schema.py();
+
+        let arguments_schema: Bound<'_, PyList> = schema.get_as_req(intern!(py, "arguments_schema"))?;
+        let mut parameters: Vec<Parameter> = Vec::with_capacity(arguments_schema.len());
+
+        let mut had_default_arg = false;
+        let mut had_keyword_only = false;
+
+        for arg in arguments_schema.iter() {
+            let arg = arg.downcast::<PyDict>()?;
+
+            let py_name: Bound<PyString> = arg.get_as_req(intern!(py, "name"))?;
+            let name = py_name.to_string();
+            let py_mode = arg.get_as::<Bound<'_, PyString>>(intern!(py, "mode"))?;
+            let py_mode = py_mode
+                .as_ref()
+                .map(|py_str| py_str.to_str())
+                .transpose()?
+                .unwrap_or("positional_or_keyword");
+
+            let mode = ParameterMode::from_str(py_mode)?;
+
+            if mode == ParameterMode::KeywordOnly {
+                had_keyword_only = true;
+            }
+
+            let schema = arg.get_as_req(intern!(py, "schema"))?;
+
+            let validator = match build_validator(&schema, config, definitions) {
+                Ok(v) => v,
+                Err(err) => return py_schema_err!("Parameter '{}':\n  {}", name, err),
+            };
+
+            let has_default = match validator {
+                CombinedValidator::WithDefault(ref v) => {
+                    if v.omit_on_error() {
+                        return py_schema_err!("Parameter '{}': omit_on_error cannot be used with arguments", name);
+                    }
+                    v.has_default()
+                }
+                _ => false,
+            };
+
+            if had_default_arg && !has_default && !had_keyword_only {
+                return py_schema_err!("Required parameter '{}' follows parameter with default", name);
+            } else if has_default {
+                had_default_arg = true;
+            }
+
+            let validation_alias = arg.get_item(intern!(py, "alias"))?;
+            let lookup_key_collection = LookupKeyCollection::new(py, validation_alias, name.as_str())?;
+
+            parameters.push(Parameter {
+                name,
+                mode,
+                lookup_key_collection,
+                validator,
+            });
+        }
+
```

Comment:
Should we be doing more here to validate the parameters make sense? e.g. `postional_or_keyword` cannot come before `positional_only`, at most one `*args` etc?